lib/badge.rb

```diff
@@ -0,0 +1,100 @@
+class Badge
+  include DSL
+
+  attr_reader :title, :description, :goal, :rank, :user
+
+  def initialize(user)
+    @user = user
+    describe_context
+    @bestowment = check_bestowment(@rank)
+  end
+
+  def check_bestowment(rank = nil)
+    bestowment = Bestowment.where(
+      badge_id: self.class,
+      user: @user
+    )
+    bestowment = bestowment.where(rank: rank) if has_progress?
```

Comment:
I know it's not clear, but each rank is a separate class, as expressed in the `DSL` class.

So if there's 5 ranks set to run `on LibraryEntry`, all 5 should execute their `#run` methods.